src/main/java/cloud/eppo/Configuration.java

```diff
@@ -0,0 +1,203 @@
+package cloud.eppo;
+
+import static cloud.eppo.Utils.getMD5Hex;
+
+import cloud.eppo.ufc.dto.*;
+import cloud.eppo.ufc.dto.adapters.EppoModule;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Encapsulates the Flag Configuration and Bandit parameters in an immutable object with a complete
+ * and coherent state.
+ *
+ * <p>A Builder is used to prepare and then create am immutable data structure containing both flag
+ * and bandit configurations. An intermediate step is required in building the configuration to
+ * accommodate the as-needed loading of bandit parameters as a network call may not be needed if
+ * there are no bandits referenced by the flag configuration.
+ *
+ * <p>Usage: Building with just flag configuration (unobfuscated is default) <code>
+ *     Configuration config = new Configuration.Builder(flagConfigJsonString).build();
+ * </code>
+ *
+ * <p>Building with bandits (known configuration) <code>
+ *     Configuration config = new Configuration.Builder(flagConfigJsonString).banditParameters(banditConfigJson).build();
+ *     </code>
+ *
+ * <p>Conditionally loading bandit models (with or without an existing bandit config JSON string).
+ * <code>
+ *  Configuration.Builder configBuilder = new Configuration.Builder(flagConfigJsonString).banditParameters(banditConfigJson);
+ *  if (configBuilder.requiresBanditModels()) {
+ *    // Load the bandit parameters encoded in a JSON string
+ *    configBuilder.banditParameters(banditParameterJsonString);
+ *  }
+ *  Configuration config = configBuilder.build();
+ * </code>
+ *
+ * <p>
+ *
+ * <p>Hint: when loading new Flag configuration values, set the current bandit models in the builder
+ * then check `requiresBanditModels()`.
+ */
+public class Configuration {
+  private static final Logger log = LoggerFactory.getLogger(Configuration.class);
+  private final Map<String, BanditReference> banditReferences;
+  private final Map<String, FlagConfig> flags;
+  private final Map<String, BanditParameters> bandits;
+  private final boolean isConfigObfuscated;
+
+  private Configuration(
+      Map<String, FlagConfig> flags,
+      Map<String, BanditReference> banditReferences,
+      Map<String, BanditParameters> bandits,
+      boolean isConfigObfuscated) {
+    this.flags = flags;
+    this.banditReferences = banditReferences;
+    this.bandits = bandits;
+    this.isConfigObfuscated = isConfigObfuscated;
+  }
+
+  public FlagConfig getFlag(String flagKey) {
```

Comment:
Most of the methods here for parsing and retrieving are cut&paste from `ConfigurationStore`